File: day01/golang/part2/part2.go
package main

import (
	"aoc/lib/jfile"
	"aoc/lib/jpy"
	"aoc/lib/jslice"
	"fmt"
	"sort"
	"strconv"
	"strings"
)

func main() {
	// fname := "example.txt"
	fname := "input.txt"

	content, _ := jfile.Read(fname)
	content = strings.TrimSpace(content)
	parts := strings.Split(content, "\n\n")
	sums := []int{}
	for _, part := range parts {
		lines := strings.Split(part, "\n")
		numbers := jpy.Map(lines, func(s string) int {
			number, _ := strconv.Atoi(s)
			return number
		})
		sums = append(sums, jpy.Sum(numbers))
	}
	sort.Ints(sums)
	jslice.Reverse(sums)

	result := jpy.Sum(sums[:3])
	fmt.Println(result)
}

File: day01/python/part2.py
#!/usr/bin/env python3

import helper


def main():
    # lines = helper.read("example.txt").split("\n\n")
    lines = helper.read("input.txt").split("\n\n")

    sums = []
    for batch in lines:
        numbers = [int(line) for line in batch.strip().split("\n")]
        sums.append(sum(numbers))
    #
    sums = sorted(sums, reverse=True)
    print(sum(sums[:3]))


##############################################################################

if __name__ == "__main__":
    main()
